packages/adapter/src/lib/adapter/adapter.ts

```diff
@@ -4459,7 +4459,14 @@ export class AdapterClass extends EventEmitter {
     findForeignObject(
         idOrName: string,
         type: ioBroker.CommonType | null,
-        options: { user?: `system.user.${string}`; language?: ioBroker.Languages },
+        options: {
+            user?: `system.user.${string}`;
+            language?: ioBroker.Languages;
+            /** Internal flag, if adapter has already checked the access rights */
+            checked?: boolean;
```

Comment:
I would propose to define two types separately ... one for the external API without "checked" and one extended type that adds this. In the API signature you use the type without checked and internally you still can define it as the internal type if relevant. The way like here I would only do if we want/need to expose the internal property officially